h5py/_hl/dataset.py

```diff
@@ -399,6 +405,10 @@ def __getitem__(self, args):
         * Boolean "mask" array indexing
         """
         args = args if isinstance(args, tuple) else (args,)
+        if is_empty_dataspace(self.id):
+            if args != tuple():
```

Comment:
I think we should also accept Ellipsis.  For example, scalar datasets can't be sliced with dset[:] but can with dset[...].